backend/tracim_backend/views/core_api/user_controller.py

```diff
@@ -607,14 +609,31 @@ def get_user_messages(
     @check_right(has_personal_access)
     @hapic.input_path(UserIdPathSchema())
     @hapic.input_headers(TracimLiveEventHeaderSchema())
-    def open_message_stream(self, context, request: TracimRequest, hapic_data=None) -> Response:
+    @hapic.input_query(TracimLiveEventQuerySchema())
+    def open_message_stream(self, context, request: TracimRequest, hapic_data) -> Response:
         """
         Open the message stream for the given user.
         Tracim Live Message Events as ServerSide Event Stream
         """
         stream_opened_event = ":Tracim Live Messages for user {}\n\nevent: stream-open\ndata:\n\n".format(
             str(request.candidate_user.user_id)
         )
+
+        after_event_id = hapic_data.query["after_event_id"]  # type: int
+        if after_event_id:
+            app_config = request.registry.settings["CFG"]  # type: CFG
+            event_api = EventApi(request.current_user, request.dbsession, app_config)
+            messages = event_api.get_messages_for_user(
+                request.candidate_user.user_id, after_event_id=after_event_id
+            )  # type: typing.List[Message]
+
+            stream_opened_event += "".join(
+                [
+                    "data:" + json.dumps(LiveMessageSchema().as_dict(message)) + "\n\n"
```

Comment:
Avoid re-creating a schema for each message, you can use one instance only. It could even be declared as a class attribute.

Reply:
Would you put this in the UserController class? This is also used for producing live message, would we store a class instance that is accessible to both the code producing the live messages and here?
Does it makes sense to actually have a ready to use class instance static `as_dict` method in LiveMessageSchema that uses a "cached" instance as a private attribute?

Reply:
Done, in LiveMessagesLib.